### ISSUE TITLE
referencePatcher.js:  value.lastIndexOf is not a function

### ISSUE THREAD

**Issue by ashleydw** [raml-org/raml-js-parser-2#457]
v1.1.1

```
TypeError: value.lastIndexOf is not a function
    at ReferencePatcher.resolveReferenceValueBasic (...\html\node_modules\raml2html\node_modules\raml-1-parser\dist\raml1\ast.core\referencePatcher.js:325:25)
    at ...\html\node_modules\raml2html\node_modules\raml-1-parser\dist\raml1\ast.core\referencePatcher.js:280:40
    at DefaultTransformer.transform (...\html\node_modules\raml2html\node_modules\raml-1-parser\dist\raml1\ast.core\expander.js:689:28)
    at ReferencePatcher.resolveReferenceValue (...\html\node_modules\raml2html\node_modules\raml-1-parser\dist\raml1\ast.core\referencePatcher.js:279:40)
    at ...\html\node_modules\raml2html\node_modules\raml-1-parser\dist\raml1\ast.core\referencePatcher.js:224:53
    at Object.visit (...\html\node_modules\raml2html\node_modules\raml-typesystem\dist\src\typeExpressions.js:90:5)
    at Object.visit (...\html\node_modules\raml2html\node_modules\raml-typesystem\dist\src\typeExpressionUtil.js:4:8)
    at ReferencePatcher.patchType (...\html\node_modules\raml2html\node_modules\raml-1-parser\dist\raml1\ast.core\referencePatcher.js:198:41)
    at ReferencePatcher.patchReferences (...\html\node_modules\raml2html\node_modules\raml-1-parser\dist\raml1\ast.core\referencePatcher.js:75:18)
    at ReferencePatcher.patchReferences (...\html\node_adules\raml2html\node_modules\raml-1-parser\dist\raml1\ast.core\referencePatcher.js:82:18)
```

If I console the value within the function; I see four types and then one included type, and then an object:

```integer
integer
string
serviceTypes.Place
PatchedReference {
  _namespace: 'serviceTypes',
  _name: 'Place',
  _collectionName: 'types',
  _referencedUnit: 
   CompilationUnit {
     _path: 'libraries/types.raml',
     _content: '....',
     _tl: true,
     _project: 
      Project {
        rootPath: '..',
        resolver: FSResolverImpl {},
        _httpResolver: [Object],
        listeners: [],
        tlisteners: [],
        pathToUnit: [Object],
        failedUnits: {},
        _fsEnabled: true,
        _namespaceResolver: [Object] },
     _apath: '..../libraries/types.raml',
     errors: [],
     _node: 
      ASTNode {
        _node: [Object],
        _unit: [Circular],
        _parent: null,
        _anchor: null,
        _include: null,
        cacheChildren: false,
        _includesContents: false,
        _errors: [],
        _highLevelParseResult: [Object],
        _highLevelNode: [Object] },
     _hl: 
      ASTNodeImpl {
        _node: [Object],
        _parent: null,
        _implicit: false,
        values: {},
        _def: [Object],
        _prop: null,
        _expanded: false,
        _allowQuestion: false,
        _subTypesCache: null,
        _auxChecked: true,
        overlayMergeMode: 0,
        _universe: [Object],
        _mergedChildren: [Object],
        _children: [Object],
        _types: [Object] } },
  _mode: 0 }
```

As a temporary fix, adding a type check at the start of `resolveReferenceValueBasic` resolves the issue:

```js
if( typeof(value) != 'string') {return value; }
```

**Comment by KonstantinSviridov** [comment]
Hi, @ashleydw 

Could you, please, share an example which causes the problem?

**Comment by ashleydw** [comment]
Looks like this was fixed in  1.1.2 but here's my minimal example for 1.1.1:

main raml file:
```
#%RAML 1.0
title: Test
version: v1
mediaType: application/json
baseUri: /api/{version}/tests

uses:
  serviceResponseTypes: ./libraries/responseTypes.raml
  serviceTypes: ./libraries/types.raml

/:
  type: { serviceResponseTypes.readOnlyCollection: { EntityType: serviceTypes.Place } }
```

./libraries/responseTypes.raml:
```
#%RAML 1.0 Library
uses:
    serviceTraits: traits.raml
resourceTypes:
  readOnlyCollection:
    get:
      is: [serviceTraits.hasResponseCollection: { EntityType: <<EntityType>> }]
```

./libraries/types.raml:
```
#%RAML 1.0 Library
types:
  Place:
    properties:
      id: number
```

**Comment by ddenisenko** [comment]
Reproducible in 1.1.1, looks fixed in 1.1.2.
